src/pipeline/src/etl/processor/letter.rs

```diff
@@ -144,6 +155,43 @@ impl Processor for LetterProcessor {
             )),
         }
     }
+
+    fn exec_mut(&self, val: &mut Vec<Value>) -> Result<(), String> {
+        for field in self.fields.iter() {
+            let index = field.input_field.index;
+            match val.get(index) {
+                Some(Value::String(s)) => {
+                    // TODO(qtang): Let this method use the intermediate state collection directly.
```

Comment:
Why not do it right now?

Reply:
This is because the original design remodeling required extensive modifications. The data structure of each processor needed to be modified. All things considered, it is ready to be implemented in the next phase.